packages/rs-drive-abci/src/abci/handlers.rs

```diff
@@ -528,48 +532,70 @@ where
 
         let RequestCheckTx { tx, .. } = request;
-        let validation_result = self.platform.check_tx(tx.as_slice())?;
+        match self.platform.check_tx(tx.as_slice()) {
+            Ok(validation_result) => {
+                let validation_error = validation_result.errors.first();
+
+                let (code, info) = if let Some(validation_error) = validation_error {
+                    let serialized_error = platform_value!(validation_error
+                        .serialize()
+                        .map_err(|e| ResponseException::from(Error::Protocol(e)))?);
+
+                    let error_data = json!({
+                        "message": "Drive check_tx error",
+                        "data": {
+                            "serializedError": serialized_error
+                        }
+                    });
+
+                    let mut error_data_buffer: Vec<u8> = Vec::new();
+                    ciborium::ser::into_writer(&error_data, &mut error_data_buffer)
+                        .map_err(|e| e.to_string())?;
 
-        let validation_error = validation_result.errors.first();
-
-        let (code, info) = if let Some(validation_error) = validation_error {
-            let serialized_error = platform_value!(validation_error
-                .serialize()
-                .map_err(|e| ResponseException::from(Error::Protocol(e)))?);
-
-            let error_data = json!({
-                "message": "Drive check_tx error",
-                "data": {
-                    "serializedError": serialized_error
-                }
-            });
-
-            let mut error_data_buffer: Vec<u8> = Vec::new();
-            ciborium::ser::into_writer(&error_data, &mut error_data_buffer)
-                .map_err(|e| e.to_string())?;
+                    (
+                        validation_error.code(),
+                        encode(&error_data_buffer, Encoding::Base64),
+                    )
+                } else {
+                    // If there are no execution errors the code will be 0
+                    (0, "".to_string())
+                };
 
-            (
-                validation_error.code(),
-                encode(&error_data_buffer, Encoding::Base64),
-            )
-        } else {
-            // If there are no execution errors the code will be 0
-            (0, "".to_string())
-        };
+                let gas_wanted = validation_result
+                    .data
+                    .map(|fee_result| fee_result.total_base_fee())
+                    .unwrap_or_default();
+                Ok(ResponseCheckTx {
+                    code,
+                    data: vec![],
+                    info,
+                    gas_wanted: gas_wanted as SignedCredits,
+                    codespace: "".to_string(),
+                    sender: "".to_string(),
+                    priority: 0,
+                })
+            }
+            Err(error) => {
+                let error_data = json!({
```

Comment:
We need to create a set of errors for ABCI (internal error, invalid request, and so on https://grpc.github.io/grpc/core/md_doc_statuscodes.html) eventually. Then we can translate them to gRPC error codes in DAPI. Not this PR of course.